presto-main/src/main/java/io/prestosql/sql/planner/LocalDynamicFilterConsumer.java

```diff
@@ -144,7 +145,13 @@ private void addPartition(TupleDomain<DynamicFilterId> tupleDomain)
             return buildChannels.keySet().stream()
                     .collect(toImmutableMap(identity(), filterId -> Domain.none(filterBuildTypes.get(filterId))));
         }
-        return result.getDomains().get();
+
+        ImmutableMap.Builder<DynamicFilterId, Domain> domains = ImmutableMap.builder();
+        domains.putAll(result.getDomains().get());
+        // Add `all` domain explicitly for dynamic filters to notify dynamic filter listeners
+        Sets.difference(buildChannels.keySet(), result.getDomains().get().keySet())
+                .forEach(filterId -> domains.put(filterId, Domain.all(filterBuildTypes.get(filterId))));
```

Comment:
maybe isntead of Sets.difference, use .putIfAbsent?

i think would be more readable